packages/block-library/src/image/edit.js

```diff
@@ -236,27 +284,129 @@ class ImageEdit extends Component {
 	}
 
 	updateAlignment( nextAlign ) {
-		const extraUpdatedAttributes = [ 'wide', 'full' ].indexOf( nextAlign ) !== -1 ?
-			{ width: undefined, height: undefined } :
-			{};
-		this.props.setAttributes( { ...extraUpdatedAttributes, align: nextAlign } );
+		if ( nextAlign === 'wide' || nextAlign === 'full' ) {
+			// Reset all sizing attributes.
+			this.resetWidthHeight();
+		}
+
+		this.props.setAttributes( { align: nextAlign } );
+	}
+
+	updateImageURL( url, imageSizeOptions ) {
+		this.resetWidthHeight();
+		let fileWidth;
+		let fileHeight;
+
+		if ( imageSizeOptions.width && imageSizeOptions.height ) {
+			fileWidth = imageSizeOptions.width;
+			fileHeight = imageSizeOptions.height;
+		} else {
+			// Find the image data.
+			map( imageSizeOptions, ( { value, imageData } ) => {
+				if ( value === url ) {
+					fileWidth = imageData.width;
+					fileHeight = imageData.height;
+					return false;
+				}
```

Comment:
using `map` without returning a new list and when modifying things inline is pretty unusual. we have a couple of choices depending on what our intention is. it looks like we're looking for a specific value and using that if it exists.

```js
updateImageURL( url, sizeOptions ) {
	const { width, height } = sizeOptions;

	if ( width && height ) {
		return this.setAttributes( {
			url,
			fileWidth: width,
			fileHeight: height,
		} );
	}

	const option = find( sizeOptions, ( { value } ) => value === url );

	if ( option ) {
		return this.setAttributes( {
			url,
			fileWidth: option.imageData.width,
			fileHeight: option.imageData.height;
		} );
	}
}
```